### PR TITLE
[LinearSolvers] Update Eigen 3.3.9

### PR DESCRIPTION
**Pull request by oberbichler**
**Description**
Update Eigen from version 3.3.7 to 3.3.~8~**9**

**Changelog**
Within Kratos: Just a copy-paste of the new files.
Within Eigen: [3.3.8-RC1](http://eigen.tuxfamily.org/index.php?title=ChangeLog#Eigen_3.3.8-rc1), [3.3.8](http://eigen.tuxfamily.org/index.php?title=ChangeLog#Eigen_3.3.8) and [3.3.9](http://eigen.tuxfamily.org/index.php?title=ChangeLog#Eigen_3.3.9)

### CODE REVIEW

**Comment by philbucher** [comment]
seems like the compiler errors come from inside the library, I also checked locally

**Comment by oberbichler** [comment]
[I think this is the bug](https://gitlab.com/libeigen/eigen/-/commit/bfdd4a9903a17ab02ed5fae34f9da8490bcb4192). Since we do not need anything from 3.3.8 I would keep this PR as a draft until there is an official fix.

**Comment by philbucher** [comment]
even compiling without MKL doesn't work for me, there is a similar issue

But I agree, lets keep this as a draft until there is an official fix

**Comment by RiccardoRossi** [comment]
why do we need this update? does it bring in important capabilities?

**Comment by oberbichler** [comment]
We do not need it.

**Comment by oberbichler** [comment]
Reminder: we should squash the commits of this pr

**Comment by philbucher** [comment]
could you please briefly summarize the changes (or copy the changelog if applicable?) Thx!

**Comment by oberbichler** [comment]
i put a link to the changelogs

**Review by philbucher — APPROVED** [review]
Guess this can go in now.

File: applications/LinearSolversApplication/external_libraries/eigen3/Eigen/Eigenvalues [hunk 1]
```diff
@@ -10,14 +10,14 @@
 
 #include "Core"
 
-#include "src/Core/util/DisableStupidWarnings.h"
```
Comment (philbucher) [inline applications/LinearSolversApplication/external_libraries/eigen3/Eigen/Eigenvalues:13]: 😂

**Pull request merged**